src/ExampleUIBar.h

```diff
@@ -3,15 +3,22 @@
 #include "Component.h"
 #include "Engine.h"
 #include "Sprite.h"
+#include "Serialization.h"
 
 class ExampleUIBar final : public Component
 {
 public:
     virtual void awake() override;
     virtual void update() override;
+    virtual void draw_editor() override;
+    float m_value = 0.0f;
+    float m_new = 0.0f;
+    NON_SERIALIZED
+    float m_value_not = 0.0f;
 
 private:
     std::shared_ptr<Entity> m_sprite_background = nullptr;
     std::shared_ptr<Entity> m_sprite_value = nullptr;
-    float m_value = 0.0f;
+
+    friend class SceneSerializer;
```

Comment:
I don't think we need this in most of the components now. This was needed because we were serializing private member variables, but now that we mostly serialize public variables, we can remove this on most of the components, apart from `Drawable` because of the `m_material`.